arm-iothub/2017-01-19/swagger/iothub.json

```diff
@@ -0,0 +1,1962 @@
+    "RoutingEventHubProperties": {
+      "description": "The properties related to an event hub endpoint.",
+      "type": "object",
+      "properties": {
+        "connectionString": {
+          "description": "The connectionstring of the event hub endpoint. ",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the event hub endpoint. name can only include alphanumeric characters, periods, underscores, hyphens with maximum length of 64 characters. The following names are reserved;  events, operationsMonitoringEvents, fileNotifications, $default. Endpoint names have to be unique across endpoint types.",
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "id": {
+          "description": "The identifier of the endpoint. An identifier will be assigned by the server, and once assigned, the identifier should never be modified. Modifying this value causes update IoT hub operations to fail. Do not pass id when specifying a new endpoint. Pass the same id that you get for an endpoint from the service when specifying an existing endpoint during update operations.",
+          "type": "string"
+        },
+        "subscriptionId": {
+          "description": "The subscription identifier of the event hub endpoint.",
+          "type": "string"
+        },
+        "resourceGroup": {
+          "description": "The name of the resource group of the event hub endpoint.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "name",
+        "connectionString"
+      ]      
+    },    
+    "RouteProperties": {
+      "description": "The properties related to a routing rule based on which the IoT hub routes messages to endpoints.",      
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the route. name can only include alphanumeric characters, periods, underscores, hyphens with maximum length of 64 characters and must be unique.",      
+          "type": "string",
+          "pattern": "^[A-Za-z0-9-._]{1,64}$"
+        },
+        "source": {
+          "description": "The source to which the routing rule is to be applied to. e.g. DeviceMessages",      
+          "enum": ["DeviceMessages"],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "RoutingSource",
+            "modelAsString": true
+          }
+        },
+        "condition": {
+          "description": "The condition which is evaluated in order to apply the routing rule. If the condition is not provided it will evaluate to true by default. For grammar, See: https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-query-language",      
+          "type": "string"
+        },
+        "endpointNames": {
+          "description": "The list of endpoints to which the messages that satisfy the condition are routed to. Currently only 1 endpoint is allowed.",
+          "minItems": 1,      
```

Comment:
We will rev the api version if it comes to that when we expand this set. But for now we will have to leave this the way it is since the RP already has these changes live.